### PR TITLE
Sanitize base_dir argument to cvmfs_server ingest

### PR DESCRIPTION
**Pull request by radupopescu**

### CODE REVIEW

**Comment by radupopescu** [comment]
@cernvm-bot cloudtest

**Comment by radupopescu** [comment]
@cernvm-bot cloudtest

File: cvmfs/swissknife_ingest.cc [hunk 1]
```diff
@@ -101,6 +101,21 @@ int swissknife::Ingest::Main(const swissknife::ArgumentList &args) {
...
 
+  // Sanitize base_directory, removing any leading or trailing slashes
+  // from non-root (!= "/") paths
+  {
+    std::string base_dir = params.base_directory;
+    if (base_dir != "/") {
+      while (HasPrefix(base_dir, "/", false) && base_dir.size() > 1) {
+        base_dir = base_dir.substr(1);
+      }
+      while (HasSuffix(base_dir, "/", false) && base_dir.size() > 1) {
```
Comment (jblomer) [inline cvmfs/swissknife_ingest.cc]: We should probably use `MakeCanonicalPath` for the trailing slash

**Comment by radupopescu** [comment]
@cernvm-bot unittest

**Pull request merged**